### DISCUSSION TITLE
load-

### DISCUSSION THREAD

**Discussion by orottler** [strongswan/strongswan#475] [Q&A]
**System (please complete the following information):**
 - OS: Debian GNU/Linux 9.13 (stretch)
 - Kernel version  4.9.0-15-amd64
 - strongSwan version(s): 5.7.2
 - Tested/confirmed with the latest version: no

**Describe the bug**
Since a long time and versions I failed with load-tester against cisco asa and get also in deep debbugging only some parameter is invalid, imm. after
charon:       06[ENC] generating IKE_AUTH request 1 [ IDi CERT N(INIT_CONTACT) CERTREQ IDr AUTH CPRQ(ADDR ADDR6 DNS DNS6) SA TSi TSr N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]

**To Reproduce**
run loadtest with

load-tester {
            # enable the plugin
            enable = yes
            fake_kernel = no
             initiators = 1
            iterations = 1
            delay = 0
            responder = 192.168.106.1
            responder_auth = pubkey
            responder_id = "C=de, , CN=moon"
            initiator_auth = pubkey
            initiator_id = 999999999992%004d
            initiator_match = * 
            initiator_tsi = %config
            initiator_tsr = 10.245.0.0/16
            issuer_key =  /etc/ipsec.d/private/load.der
            digest = sha384
            issuer_cert = /etc/ipsec.d/aacerts/caCert.der
            ca_dir =  /etc/ipsec.d/cacerts/
              proposal = aes256-sha256-prfsha1-modp2048
            esp = aes256-sha256-modp2048
            request_virtual_ip = yes
            ike_rekey = 172800
            # enable CHILD_SA every 3600s
            child_rekey = 3600
            delete_after_established = no
            shutdown_when_complete = no

        }

**Expected behavior**
should establish a connection like the simple peer connection

conn asa0
    right=192.168.106.1
    rightid="C=de, L=DATEV-Zentrale, O=datev, CN=dnet.vpn.qs.datev.de, unstructuredName=risavpn01-qs.infra.dvint.de"
    rightsubnet=10.245.0.0/16
    rightauth=pubkey
    left=192.168.106.3
    leftsourceip=%config
    leftcert=/etc/ipsec.d/acerts/peer.der
    auto=add
    keyexchange=ikev2
    ike=aes256-sha256-prfsha1-modp2048
    # CHILD_SA proposal to use for load tests.
    esp=aes256-sha256-modp2048
    modeconfig = push
    ikelifetime = 172800s
    keylife = 3600s
    mobike = no

**Logs/Backtraces**
If applicable, add logs or backtraces to help explain your problem.
Jun 23 12:17:16 risasrv01-qs charon: 05[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(HASH_ALG) N(REDIR_SUP) ]
Jun 23 12:17:16 risasrv01-qs charon: 05[NET] sending packet: from 192.168.106.3[500] to 192.168.106.1[500] (456 bytes)
Jun 23 12:17:16 risasrv01-qs charon: 08[NET] received packet: from 192.168.106.1[500] to 192.168.106.3[500] (719 bytes)
Jun 23 12:17:16 risasrv01-qs charon: 08[ENC] parsed IKE_SA_INIT response 0 [ SA KE No V V N(NATD_S_IP) N(NATD_D_IP) CERTREQ N(FRAG_SUP) V ]
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] received Cisco Delete Reason vendor ID
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] received Cisco Copyright (c) 2009 vendor ID
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] received FRAGMENTATION vendor ID
Jun 23 12:17:16 risasrv01-qs charon: 08[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA1/MODP_2048
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] received cert request for "C=DE, *****"
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] received cert request for "C=DE, ****"
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] received 5 cert requests for an unknown ca
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] sending cert request for "C=DE, ****"
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] sending cert request for "C=DE, ****"
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] authentication of '9999999999920001' (myself) with RSA signature successful
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] sending end entity cert "CN=9999999999920001"
Jun 23 12:17:16 risasrv01-qs charon: 08[IKE] establishing CHILD_SA load-test{1}
Jun 23 12:17:16 risasrv01-qs charon: 08[ENC] generating IKE_AUTH request 1 [ IDi CERT N(INIT_CONTACT) CERTREQ IDr AUTH CPRQ(ADDR ADDR6 DNS DNS6) SA TSi TSr N(EAP_ONLY) N(MSG_ID_SYN_SUP) ]
Jun 23 12:17:16 risasrv01-qs charon: 08[NET] sending packet: from 192.168.106.3[500] to 192.168.106.1[500] (2368 bytes)
Jun 23 12:17:20 risasrv01-qs charon: 10[IKE] retransmit 1 of request with message ID 1
Jun 23 12:17:20 risasrv01-qs charon: 10[NET] sending packet: from 192.168.106.3[500] to 192.168.106.1[500] (2368 bytes)

**Additional context**

Don't know some parameter are not available for loadtest (modeconfig ?) , Additional there seems a new dual stack option. I provided an IPv6 pool/DNS6 servers  too on the gateway , but it doesnt help.  Thats the only difference I see regarding the working single peer config

**Comment by tobiasbrunner** [comment]
I don't see any error. Either there is a network issue or the peer decides to not respond on purpose. Maybe an IP fragmentation problem due to the size of the IKE_AUTH message (although this is local, so kinda weird that fragments would get dropped). Firewall problem (unless it's related to the mentioned fragments) would also seem weird as there is no port switch from 500 to 4500 (MOBIKE is disabled for load-tester configs and there apparently is no NAT, so that's fine). So maybe check firewall rules regarding IP fragments. Or read the log of the other peer in case it actually receives the IKE_AUTH message but decided not to respond for some reason.
